etna/models/nn/chronos/base.py

```diff
@@ -0,0 +1,273 @@
+import os
+import warnings
+import zipfile
+from abc import abstractmethod
+from pathlib import Path
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Union
+from urllib import request
+
+import pandas as pd
+
+from etna import SETTINGS
+from etna.datasets import TSDataset
+from etna.distributions import BaseDistribution
+from etna.libs.chronos import BaseChronosPipeline
+from etna.libs.chronos import ChronosBoltModelForForecasting
+from etna.libs.chronos import ChronosModelForForecasting
+from etna.models.base import PredictionIntervalContextRequiredAbstractModel
+
+if SETTINGS.chronos_required:
+    import torch
+
+
+class ChronosBaseModel(PredictionIntervalContextRequiredAbstractModel):
+    """Base class for Chronos-like pretrained models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        encoder_length: int,
+        device: str,
+        dtype: torch.dtype,
+        cache_dir: Path,
+        from_s3: bool,
+    ):
+        """
+        Init Chronos-like model.
+
+        Parameters
+        ----------
+        model_name:
+            Model name. See ``pretrained_model_name_or_path`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        encoder_length:
+            Number of last timestamps to use as a context.
+        device:
+            Device type. See ``device_map`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        dtype:
+            Torch dtype of computation. See ``torch_dtype`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        cache_dir:
+            Local path to save model from huggingface during first model initialization. All following class initializations appropriate model version will be downloaded from this path.
+            See ``cache_dir`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        from_s3:
+            Whether to load from s3 or huggingface. Mostly for developer usage.
+        """
+        super().__init__()
+        self.model_name = model_name
+        self.encoder_length = encoder_length
+        self.device = device
+        self.dtype = dtype
+        self.cache_dir = cache_dir
+        self.from_s3 = from_s3
+
+        if self.from_s3:
```

Comment:
It looks weird, I don't really like how it works right now:
- It doesn't have any flexibility for a developer: the model names and s3 bucket are hard-coded
- This parameter is too public for "developer-only" use-cases

Reply:
- Why do you need to rename model names and s3 bucket? In addition to the discussion below I think we will touch models to change, finetune them or something else with a very low probability.
- Any ideas?

Reply:
Can't we check smth like this: `https://s3fs.readthedocs.io/en/latest/#integration` ?

Reply:
Look at `filepath_or_buffer` parameter [here](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).

Reply:
Ok
As we have 2 options of path (s3 and local/hf), we can simply parse by hands what kind of path it is.

Reply:
- We should describe such possibility in parameter docs
- We should somehow explain how secrets for s3 could be set (probably like in any other place in the library)